build/templates/_attributes.py.mako

```diff
@@ -104,6 +105,38 @@ class AttributeEnum(object):
         return self._underlying_attribute.__set__(session, value.value)
 
 
+class AttributeEnumWithConverter(AttributeEnum):
+    '''Class for attributes that use enums internally but are exposed in nimi-python API as something else, thus need conversion.'''
+
+    def __init__(self, underlying_enum_attribute, getter_converter, setter_converter):
+        '''Creates and returns an instance of AttributeEnumWithConverter attribute meta class.
+
+        Args:
+            underlying_enum_attribute (AttributeEnum): The AttributeEnum instance for the underlying
+                enum
+
+            getter_converter (function): The function that converts the enum value to its converted
+                value
+
+            setter_converter (function): The function that converts the converted value back to the
+                enum value
+        '''
+        self._underlying_enum_attribute = underlying_enum_attribute
+        self._getter_converter = getter_converter
+        self._setter_converter = setter_converter
+
+    def __get__(self, session, session_type):
+        try:
+            return self._getter_converter(
+                self._underlying_enum_attribute.__get__(session, session_type)
+            )
+        except ValueError:
+            raise errors.DriverTooNewError('The driver runtime returned an unexpected value. ')
+
+    def __set__(self, session, value):
+        return self._underlying_enum_attribute.__set__(session, self._setter_converter(value))
```

Comment:
I think you need a try/except in here.

@sbethur this is a place where being restrictive can come back and bite us. For example if an external or internal client is using the Python API with unreleased driver or using a private value... we are going to prevent it.

Are we better off in those cases just passing the unknown values up and down to the driver runtime and letting it error? If that's the case then the new DriverTooNewError is not needed after all. Let's discuss.